providers/implementations/ciphers/ciphercommon_ccm.c

```diff
@@ -109,7 +109,10 @@ int ossl_ccm_set_ctx_params(void *vctx, const OSSL_PARAM params[])
             ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_IV_LENGTH);
             return 0;
         }
-        ctx->l = ivlen;
+        if (ctx->l != ivlen) {
+            ctx->l = ivlen;
+            ctx->iv_set = 0;
```

Comment:
Is this  tested-  what happens with this line.... ```` if (!ctx->len_set && !ccm_set_iv(...)````)

Reply:
Same comment appllies to gcm

Reply:
i.e. Is the test falling over for the reason you think it is.

Reply:
I am sorry, it is totally unclear to me what do you mean. The test fails without the change.